pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/TableRebalanceIntegrationTest.java

```diff
@@ -88,6 +88,78 @@ private String getRebalanceUrl(RebalanceConfig rebalanceConfig, TableType tableT
         + "?type=" + tableType.toString() + "&" + getQueryString(rebalanceConfig);
   }
 
+  @Test
+  public void testImplicitRealtimeTableInstanceAssignment() throws Exception {
```

Comment:
Great test to have! Would it also make sense to add a test to `TableRebalancerClusterStatelessTest` and actually validate the segment assignment before and after rebalance for this instance selector? (similar to your instance selector test, but with an actual rebalance in the mix)?